atlasdb-cassandra/src/main/java/com/palantir/atlasdb/cassandra/CassandraServersConfigs.java

```diff
@@ -123,22 +121,20 @@ final void check() {
 
         public abstract Set<InetSocketAddress> cqlHosts();
 
-        public abstract Optional<SocketAddress> socksProxy();
-
         @Value.Default
         public CqlCapableConfigTuning tuning() {
             return ImmutableCqlCapableConfigTuning.builder().build();
         }
 
         @Override
-        public final int numberOfThriftHosts() {
+        public final int numberOfHosts() {
+            Preconditions.checkState(thriftHosts().size() == cqlHosts().size());
             return thriftHosts().size();
         }
```

Comment:
Discussed offline:
- This shouldn't throw violently, so `Preconditions.checkState(...)` is too heavy handed.
- This check is now only done when you call `numberOfHosts`, not at object creation time - should validate that this is indeed the intention.